src/StepAggregator.php

```diff
@@ -195,4 +182,26 @@ private function buildPipeline()
 
         return $nextCallable;
     }
+
+    /**
+     * Sorts the internal list of steps and writers by priority.
+     *
+     * @return Step[]
+     */
+    private function getStepsSortedByPriority()
+    {
+        $steps = $this->steps;
+        // Use illogically large and small priorities
+        $steps[-255][] = new Step\ArrayCheckStep;
+        foreach ($this->writers as $writer) {
+            $steps[-256][] = new Step\WriterStep($writer);
+        }
+
+        ksort($steps);
+        $stepsIterator = new \RecursiveIteratorIterator(
+            new \RecursiveArrayIterator($steps, \RecursiveArrayIterator:: CHILD_ARRAYS_ONLY)
+        );
+
+        return iterator_to_array($stepsIterator, false);
```

Comment:
This looks strange, we have an array than we create a iterator with it and 2 lines later we transform the iterator back to an array.

Reply:
@Baachi I changed the implementation to use only array.